python/ray/util/spark/data.py

```diff
@@ -0,0 +1,46 @@
+import ray
+from urllib.parse import urlparse
+from ray.util.spark.utils import get_spark_session, is_in_databricks_runtime
+
+
+def _convert_dbfs_path_to_local_path(dbfs_path):
+    parsed_path = urlparse(dbfs_path)
+    assert parsed_path.scheme.lower() == "dbfs", "dbfs path is required."
+    return "/dbfs" + parsed_path.path
+
+
+def load_spark_dataset(path, saved_format):
```

Comment:
Shouldn't this be added to Ray Dataset API directly? cc @clarkzinzow @c21 for discussion.

Reply:
This sounds like a good idea

Reply:
CC @clarkzinzow @c21 Thoughts ?

Reply:
Hmm, it seems like the main thing this is providing is delta compatibility. If we want to add it natively, it could be something like ray.data.read_delta().

For supporting dbfs:, I think that could be an thin arrow filesystem impl, so the final call could be ray.data.read_delta(..., filesystem=DatabricksFileSystem()). The filesystem could just translate paths and delegate to the local arrow filesystem.

Reply:
>  it seems like the main thing this is providing is delta compatibility. If we want to add it natively, it could be something like ray.data.read_delta().

@ericl 

Delta format saved data are also parquet format files, (but with some additional logs).
So read parquets can directly support read delta data. Do we need another API to do that ?

For apache/spark, by default the DataFrame is saved in parquet format, 
but in databricks runtime, by default the DataFrame is saved in delta format, 

so if the Ray loader API can automatically support parquet/delta format, it will be more friendly to user, user don't need to check whether the saved data is parquet or delta format.

Reply:
Btw, we also want to support load data from "spark table" via the same API,
i.e. the `load_spark_dataset(source)`, source can be either a spark dataset path (parquet or delta format), or a spark table name. 

In this case, shall we still move this method to `ray.data` ? Its interface is not like other APIs in `ray.data`.